\PassOptionsToPackage{unicode=true}{hyperref} % options for packages loaded elsewhere
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provides euro and other symbols
\else % if luatex or xelatex
  \usepackage{unicode-math}
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage[]{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\usepackage{hyperref}
\hypersetup{
            pdftitle={Proyecto Copepodos},
            pdfauthor={Javier Urban},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother


\title{Proyecto Copepodos}
\author{Javier Urban}
\date{}

\begin{document}
\maketitle

\hypertarget{proyecto-de-structura-genetica-de-cuatro-poblaciones-de-copuxe9podos}{%
\subsubsection{Proyecto de structura genetica de cuatro poblaciones de
Copépodos}\label{proyecto-de-structura-genetica-de-cuatro-poblaciones-de-copuxe9podos}}

Cargár paquetes necesarios para los análisis

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(adegenet)}
\KeywordTok{library}\NormalTok{(vcfR)}
\KeywordTok{library}\NormalTok{(poppr)}
\KeywordTok{library}\NormalTok{(ape)}
\KeywordTok{library}\NormalTok{(RColorBrewer)}
\end{Highlighting}
\end{Shaded}

\hypertarget{crear-archivo-que-incluya-datos-geneticos-y-meta-datos}{%
\subsection{Crear archivo que incluya datos geneticos y meta
datos}\label{crear-archivo-que-incluya-datos-geneticos-y-meta-datos}}

Se carga el archivo .VCF con 11254 SNPs y la información de las
poblaciones del copépodo \emph{Leptodiaptomus} a un archivo genligth
para análisis de divergencia genética

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{copepods.vcf <-}\StringTok{ }\KeywordTok{read.vcfR}\NormalTok{(}\StringTok{"../data/out.recode.vcf"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Scanning file to determine attributes.
## File attributes:
##   meta lines: 74
##   header_line: 75
##   variant count: 11254
##   column count: 102
## Meta line 74 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 11254
##   Character matrix gt cols: 102
##   skip: 0
##   nrows: 11254
##   row_num: 0
## Processed variant 1000Processed variant 2000Processed variant 3000Processed variant 4000Processed variant 5000Processed variant 6000Processed variant 7000Processed variant 8000Processed variant 9000Processed variant 10000Processed variant 11000Processed variant: 11254
## All variants processed
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{copepods.vcf}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## ***** Object of Class vcfR *****
## 93 samples
## 14 CHROMs
## 11,254 variants
## Object size: 128 Mb
## 0 percent missing data
## *****        *****         *****
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pop_names <-}\StringTok{ }\KeywordTok{read.table}\NormalTok{(}\StringTok{"../data/pop_nom_copepods.txt"}\NormalTok{, }\DataTypeTok{sep =} \StringTok{"}\CharTok{\textbackslash{}t}\StringTok{"}\NormalTok{, }\DataTypeTok{header =} \OtherTok{TRUE}\NormalTok{)}
\KeywordTok{all}\NormalTok{(}\KeywordTok{colnames}\NormalTok{(copepods.vcf}\OperatorTok{@}\NormalTok{gt)[}\OperatorTok{-}\DecValTok{1}\NormalTok{]}\OperatorTok{==}\NormalTok{pop_names}\OperatorTok{$}\NormalTok{Individula.ID)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] TRUE
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{##Create genligth object with SNPs and delimited populations to use adegenet and poppr}
\NormalTok{gl.copepods <-}\StringTok{ }\KeywordTok{vcfR2genlight}\NormalTok{(copepods.vcf)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in vcfR2genlight(copepods.vcf): Found 2564 loci with more than two alleles.
## Objects of class genlight only support loci with two alleles.
## 2564 loci will be omitted from the genlight object.
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{ploidy}\NormalTok{(gl.copepods) <-}\StringTok{ }\DecValTok{2}
\KeywordTok{pop}\NormalTok{(gl.copepods) <-}\StringTok{ }\NormalTok{pop_names}\OperatorTok{$}\NormalTok{Pop}
\NormalTok{gl.copepods}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  /// GENLIGHT OBJECT /////////
## 
##  // 93 genotypes,  8,690 binary SNPs, size: 1.1 Mb
##  0 (0 %) missing data
## 
##  // Basic content
##    @gen: list of 93 SNPbin
##    @ploidy: ploidy of each individual  (range: 2-2)
## 
##  // Optional content
##    @ind.names:  93 individual labels
##    @loc.names:  8690 locus labels
##    @chromosome: factor storing chromosomes of the SNPs
##    @position: integer storing positions of the SNPs
##    @pop: population of each individual (group size range: 22-24)
##    @other: a list containing: elements without names
\end{verbatim}

\hypertarget{anuxe1lisis-de-divergencia-gnuxe9tica-del-copuxe9podo-leptodiaptomus}{%
\subsection{\texorpdfstring{Análisis de divergencia gnética del copépodo
\emph{Leptodiaptomus}}{Análisis de divergencia gnética del copépodo Leptodiaptomus}}\label{anuxe1lisis-de-divergencia-gnuxe9tica-del-copuxe9podo-leptodiaptomus}}

Se calculó la distancia genética entre individuos y poblaciones para su
posterior uso en análisis de estructura y divergencia genética

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{copes.dis <-}\StringTok{ }\KeywordTok{dist}\NormalTok{(gl.copepods)}
\NormalTok{copes.dis <-}\StringTok{ }\NormalTok{poppr}\OperatorTok{::}\KeywordTok{bitwise.dist}\NormalTok{(gl.copepods) }
\end{Highlighting}
\end{Shaded}

\hypertarget{uxe1rbol-filogenuxe9tico-basado-en-distancia-genuxe9tica-para-4-poblaciones-del-copuxe9podo-leptodiaptomus}{%
\subsection{\texorpdfstring{Árbol filogenético basado en distancia
genética para 4 poblaciones del copépodo
\emph{Leptodiaptomus}}{Árbol filogenético basado en distancia genética para 4 poblaciones del copépodo Leptodiaptomus}}\label{uxe1rbol-filogenuxe9tico-basado-en-distancia-genuxe9tica-para-4-poblaciones-del-copuxe9podo-leptodiaptomus}}

El árbol se genero basado en la distancia bitwise.dist y un método
``upgma''

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{PhyloCopes <-}\StringTok{ }\KeywordTok{aboot}\NormalTok{(gl.copepods, }\DataTypeTok{tree =} \StringTok{"upgma"}\NormalTok{, }\DataTypeTok{distance =}\NormalTok{ bitwise.dist, }\DataTypeTok{sample =} \DecValTok{100}\NormalTok{, }\DataTypeTok{showtree =}\NormalTok{ F, }\DataTypeTok{cutoff =} \DecValTok{50}\NormalTok{, }\DataTypeTok{quiet =}\NormalTok{ T)}
\NormalTok{cols <-}\StringTok{ }\KeywordTok{brewer.pal}\NormalTok{(}\DataTypeTok{n =} \KeywordTok{nPop}\NormalTok{(gl.copepods), }\DataTypeTok{name =} \StringTok{"Dark2"}\NormalTok{)}
\KeywordTok{plot.phylo}\NormalTok{(PhyloCopes, }\DataTypeTok{cex =} \FloatTok{0.9}\NormalTok{, }\DataTypeTok{font =} \DecValTok{2}\NormalTok{, }\DataTypeTok{adj =} \DecValTok{0}\NormalTok{, }\DataTypeTok{tip.color =}\NormalTok{ cols[}\KeywordTok{pop}\NormalTok{(gl.copepods)])}
\KeywordTok{nodelabels}\NormalTok{(PhyloCopes}\OperatorTok{$}\NormalTok{node.label, }\DataTypeTok{adj =} \KeywordTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{, }\FloatTok{-0.4}\NormalTok{), }\DataTypeTok{frame =} \StringTok{"n"}\NormalTok{, }\DataTypeTok{xpd=}\OtherTok{TRUE}\NormalTok{)}
\KeywordTok{legend}\NormalTok{(}\StringTok{'topleft'}\NormalTok{, }\DataTypeTok{legend =} \KeywordTok{c}\NormalTok{(}\StringTok{"ATX"}\NormalTok{, }\StringTok{"CAR"}\NormalTok{, }\StringTok{"PRE"}\NormalTok{, }\StringTok{"QUE"}\NormalTok{), }\DataTypeTok{fill =}\NormalTok{ cols, }\DataTypeTok{border =} \OtherTok{TRUE}\NormalTok{, }\DataTypeTok{bty =} \StringTok{"n"}\NormalTok{, }\DataTypeTok{cex =} \DecValTok{1}\NormalTok{)}
\KeywordTok{axis}\NormalTok{(}\DataTypeTok{side =} \DecValTok{1}\NormalTok{)}
\KeywordTok{title}\NormalTok{(}\DataTypeTok{xlab =} \StringTok{"Distancia genética"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Proyecto-copépodos_files/figure-latex/unnamed-chunk-4-1.pdf}

\hypertarget{anuxe1lisis-de-estructura-genuxe9tica}{%
\subsection{Análisis de estructura
genética}\label{anuxe1lisis-de-estructura-genuxe9tica}}

Se generó una red de expansión mínima \emph{MSN} para visualizar la
estructura genética

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(igraph)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Attaching package: 'igraph'
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:ape':
## 
##     edges, mst, ring
\end{verbatim}

\begin{verbatim}
## The following objects are masked from 'package:stats':
## 
##     decompose, spectrum
\end{verbatim}

\begin{verbatim}
## The following object is masked from 'package:base':
## 
##     union
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{copes.msn <-}\StringTok{ }\KeywordTok{poppr.msn}\NormalTok{(gl.copepods, copes.dis, }\DataTypeTok{showplot =} \OtherTok{FALSE}\NormalTok{, }\DataTypeTok{include.ties =}\NormalTok{ T)}
\NormalTok{node.size <-}\StringTok{ }\KeywordTok{rep}\NormalTok{(}\DecValTok{2}\NormalTok{, }\DataTypeTok{times =} \KeywordTok{nInd}\NormalTok{(gl.copepods))}
\KeywordTok{names}\NormalTok{(node.size) <-}\StringTok{ }\KeywordTok{indNames}\NormalTok{(gl.copepods)}
\KeywordTok{vertex_attr}\NormalTok{(copes.msn}\OperatorTok{$}\NormalTok{graph)}\OperatorTok{$}\NormalTok{size <-}\StringTok{ }\NormalTok{node.size}
\KeywordTok{set.seed}\NormalTok{(}\DecValTok{9}\NormalTok{)}
\KeywordTok{plot_poppr_msn}\NormalTok{(gl.copepods, copes.msn, }\DataTypeTok{palette =}\NormalTok{ cols)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Proyecto-copépodos_files/figure-latex/unnamed-chunk-5-1.pdf}

\hypertarget{analysis-of-pca}{%
\subsection{Analysis of PCA}\label{analysis-of-pca}}

Se calculan los valores propios para generar el número de PC que
expliquen el mayor porcentaje de variación

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{copes.pca <-}\StringTok{ }\KeywordTok{glPca}\NormalTok{(gl.copepods, }\DataTypeTok{nf =} \DecValTok{3}\NormalTok{)}
\KeywordTok{barplot}\NormalTok{(}\DecValTok{100}\OperatorTok{*}\NormalTok{copes.pca}\OperatorTok{$}\NormalTok{eig}\OperatorTok{/}\KeywordTok{sum}\NormalTok{(copes.pca}\OperatorTok{$}\NormalTok{eig), }\DataTypeTok{col =} \KeywordTok{heat.colors}\NormalTok{(}\DecValTok{50}\NormalTok{), }\DataTypeTok{main=}\StringTok{"PCA Eigenvalues"}\NormalTok{)}
\KeywordTok{title}\NormalTok{(}\DataTypeTok{ylab=}\StringTok{"Porcentaje"}\NormalTok{)}
\KeywordTok{title}\NormalTok{(}\DataTypeTok{xlab =} \StringTok{"Eigenvalues"}\NormalTok{, }\DataTypeTok{line =} \DecValTok{1}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Proyecto-copépodos_files/figure-latex/unnamed-chunk-6-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{copes.pca.scores <-}\StringTok{ }\KeywordTok{as.data.frame}\NormalTok{(copes.pca}\OperatorTok{$}\NormalTok{scores)}
\NormalTok{copes.pca.scores}\OperatorTok{$}\NormalTok{pop <-}\StringTok{ }\KeywordTok{pop}\NormalTok{(gl.copepods)}
\end{Highlighting}
\end{Shaded}

Graficas de la distribución de los datos en el PC1 y PC2

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{library}\NormalTok{(ggplot2)}
\KeywordTok{set.seed}\NormalTok{(}\DecValTok{9}\NormalTok{)}
\NormalTok{plotpca <-}\StringTok{ }\KeywordTok{ggplot}\NormalTok{(copes.pca.scores, }\KeywordTok{aes}\NormalTok{(}\DataTypeTok{x=}\NormalTok{PC1, }\DataTypeTok{y=}\NormalTok{PC2, }\DataTypeTok{colour=}\NormalTok{pop))}\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_point}\NormalTok{(}\DataTypeTok{size=}\DecValTok{3}\NormalTok{)}\OperatorTok{+}\StringTok{ }
\StringTok{  }\KeywordTok{scale_color_manual}\NormalTok{(}\DataTypeTok{values=}\NormalTok{cols)}\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_hline}\NormalTok{(}\DataTypeTok{yintercept=}\DecValTok{0}\NormalTok{)}\OperatorTok{+}
\StringTok{  }\KeywordTok{geom_vline}\NormalTok{(}\DataTypeTok{xintercept=}\DecValTok{0}\NormalTok{)}\OperatorTok{+}
\StringTok{  }\KeywordTok{theme_bw}\NormalTok{()}
\NormalTok{plotpca}
\end{Highlighting}
\end{Shaded}

\includegraphics{Proyecto-copépodos_files/figure-latex/unnamed-chunk-7-1.pdf}

Graficas de los PCA que integra la probabilidad de pertenencia a la
población para cada muestra a cada una de las poblaciones
predeterminadas

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{##DAPC graphic}
\NormalTok{copes.dapc <-}\StringTok{ }\KeywordTok{dapc}\NormalTok{(gl.copepods, }\DataTypeTok{n.pca =} \DecValTok{3}\NormalTok{, }\DataTypeTok{n.da =} \DecValTok{2}\NormalTok{)}
\KeywordTok{scatter}\NormalTok{(copes.dapc, }\DataTypeTok{col=}\NormalTok{cols, }\DataTypeTok{cex=}\DecValTok{3}\NormalTok{, }\DataTypeTok{legend =} \OtherTok{TRUE}\NormalTok{, }\DataTypeTok{clabel=}\NormalTok{F, }\DataTypeTok{posi.leg =} \StringTok{"bottomleft"}\NormalTok{, }\DataTypeTok{scree.pca =} \OtherTok{TRUE}\NormalTok{, }\DataTypeTok{posi.pca =} \StringTok{"topright"}\NormalTok{, }\DataTypeTok{cleg =} \FloatTok{0.5}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Proyecto-copépodos_files/figure-latex/unnamed-chunk-8-1.pdf}

\end{document}
